### DISCUSSION TITLE
GraphQL

### DISCUSSION THREAD

**Discussion by ricardo-valero** [solidjs-community/solid-primitives#55] [Ideas]
I just saw the `@solid-primitives/graphql` I'll test it out soon

But is it http only? 
Does it support subscriptions? SSE? ws?

I also like this approach [GQty](https://gqty.dev/docs/intro) for writing queries instead of the using strings

**Comment by thetarnav** [comment]
Our `graphql` package is right now very primitive, so there aren't any advanced features, besides simple fetches.
We probably won't be able build a fully fledged gql library in the near future, but we'll keep adding the most important stuff, piece by piece.
If you have some ideas of what these priorities should be, then let us know, or help us build them 😁  
Anyway, I wont be able to discuss with you about gql, as I know little about it. The idea of abstracting the query strings with typescript is very interesting though. Will definitely take a look at this later.

**Comment by davedbase** [comment]
I'll second @thetarnav comments on this. This primitive came from a very humble origins, ported from a core community members package. I personally will be working on a number of GQL oriented services on my next Solid project so there will likely be an opportunity to improve it. It's funny that you mention SSE and WS because I'll have needs for that specifically.

GQty is a really neat approach but our primitives will only ever focus on the mechanic for handling GQL requests and responses within a reactive context. This might mean that we'll expand the primitives to do slightly more, perhaps I can add an SSE and WS option too. We have other primitives that can support that. The goal for these primitives are to be foundational and highly composable.

What we wont do is adopt a pattern like GQty however, we are strongly considering creating a segment of Solid Primitives for vendor integrations. We might call it Solid Integrations where we can house and maintain Solid enabled adapters for projects like urql, GQty and others. If you're interested in helping out definitely let us know!

The point here is we want Solid Primitives to be more about the tools and Solid Integrations for the more opinionated parts.

**Reply by ricardo-valero** [reply]
I'm very interested but just learned about Solid some weeks ago.
Right now I'm just poking around. But definitely forward to see something with SSE and WS 😄 

> our primitives will only ever focus on the mechanic for handling GQL requests and responses

I agree, I got sidetracked mentioning GQty.

**Reply by trusktr** [reply]
I just took a look at GQty, and actually the API feels like Meteor collection queries, and could totally be reactive. For example, in

```js
 return <img src={query.user({ id }).avatarUrl({ size: 100 })} />
```

the `query.user({ id }).avatarUrl({ size: 100 })` can be a reactive expression, usable in any reactive context (effect). In a sense it would similar to accessing a store as in `store.user[id].avatarUrl`.

**Comment by BierDav** [comment]
Can we use `@graphql-codegen` on par with `@solid-primitives/graphql`? If so, does anyone know how to do so?
